src/controllers/cart.js

```diff
@@ -19,18 +20,18 @@ exports.postCart = async (req, res, next) => {
     await cart.save();
     return res.status(201).json(cart);
   } catch (err) {
-    return next(new CustomError(err.message, 500));
+    return next(new CustomError(err.message));
   }
 };
 
-exports.deleteCart = async (req, res) => {
+exports.deleteCart = async (req, res, next) => {
   try {
     Cart.deleteOne({ customerId: req.user.id });
-    res.status(204).json({
+    return res.status(204).json({
```

Comment:
If the status code is 204, it means nothing will be displayed in the response body
so instead use this: ```return res.sendStatus(204);```